esp-hal/src/spi/master.rs

```diff
@@ -115,42 +112,43 @@ const MAX_DMA_SIZE: usize = 32736;
 ///
 /// Used to define specific commands sent over the SPI bus.
 /// Can be [Command::None] if command phase should be suppressed.
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub enum Command {
```

Comment:
Not 100% happy with this. We have limited options, i.e. command has to be narrower or equal than address which may be 1 or N bits depending on the data mode. We don't have the freedom of configurability implied by separate SpiDataMode options.

Reply:
The current hardware basically limits us to these:

```rust
enum TransferMode {
    Spi,
    Qpi, // 4 bit command, address, data
    Opi, // 8 bit command, address, data
    QspiNarrowAddress(1/2/4 data bits), // 1 bit command, 1 bit address
    QspiWideAddress(1/2/4 data bits),   // 1 bit command, N bit address
}
```